docs/apis/es_reader.md

```diff
@@ -0,0 +1,137 @@
+
+
+## Elasticsearch Reader Factory API Methods
+
+### size
+
+this will return how many seperate reader apis are in the cache
+
+### get
+parameters:
+- name: String
+
+this will fetch any reader api that is associated with the name provided
+
+### getConfig
+parameters:
+- name: String
+
+this will fetch any reader api config that is associated with the name provided
+
+### create (async)
+parameters:
+- name: String
+- configOverrides: Check options below, optional
+
+this will create an instance of a reader api, and cache it with the name given. Any config provided in the second argument will override what is specified in the apiConfig and cache it with the name provided. It will throw an error if you try creating another api with the same name parameter
+
+```typescript
+    const apiManager = this.getAPI<ElasticReaderFactoryAPI>(apiName);
+    // this will return an api cached at "normalClient" and this instance will use all configurations listed on the apiConfig
+    const client = apiManager.create('normalClient')
+
+    // this will return an api cached at "overrideClient" and this instance will have an override setting the parameter index to "other_index", this will use the rest of the configurations listed in the apiConfig
+    const overrideClient = apiManager.create('overrideClient', { index: 'other_index'})
+```
+
+### remove (async)
+parameters:
+- name: String
+
+this will remove an instance of a reader api from the cache and will follow any cleanup code specified in the api code.
+
+### entries
+
+This will allow you to iterate over the cache name and client of the cache
+
+### keys
+
+This will allow you to iterate over the cache name of the cache
+
+### values
+
+This will allow you to iterate over the clients of the cache
+
+
+## Elasticsearch Reader Instance
+This is the reader class that is returned from the create method of the APIFactory. This returns an [elastic-api](https://terascope.github.io/teraslice/docs/packages/elasticsearch-api/overview).
+
+### search
+```(query: ElasticsearchQuery) => Promise<DataEntities[]>```
+parameters:
+- query: an elasticsearch query object
+
+### count
+```(query: ElasticsearchQuery) => Promise<number>```
+parameters:
+- query: an elasticsearch query object
+
+
+```js
+    const query: {
+        q: '(test:query OR other:thing AND bytes:>=2000)',
+        size: 100,
+        fields: 'foo,bar,date'
+    };
+
+    const results = await api.search(query)
+```
+
+## Options
+
+| Configuration | Description | Type |  Notes   |
+| --------- | -------- | ------ | ------ |
+| \_name | Name of operation, it must reflect the exact name of the file | String | required |
+| endpoint | The base API endpoint to read from: i.e.http://yourdomain.com/api/v1 | String | required |
+| token | teraserver API access token for making requests | String | required |
```

Comment:
Does it really have a `token`? This seems like it is a copy and paste from the `spaces_reader_api`?